### PR TITLE
Row-based isSqlNull()

### PR DESCRIPTION
**Pull request by isoos**
- https://github.com/isoos/postgresql-dart/issues/322
- alternative to #323

### CODE REVIEW

**Comment by davidmartos96** [comment]
This approach is even easier. Nice job!

File: lib/postgres.dart [hunk 1]
```diff
@@ -262,11 +262,17 @@ class Result extends UnmodifiableListView<ResultRow> {
...
 
   ResultRow({
     required List<Object?> values,
     required this.schema,
-  }) : super(values);
+    List<bool>? sqlNulls,
+  })  : _sqlNulls = sqlNulls,
+        super(values);
+
+  /// Returns true if the result at [columnIndex] returned SQL `NULL` value.
```
Comment (simolus3) [inline lib/postgres.dart:274]: Users might be surprised about the distinction as well, so I suggest we explain it:

```suggestion
  /// Returns true if the result at [columnIndex] returned SQL `NULL` value.
  ///
  /// This is not necessarily the same as `this[columnIndex] == null`: For
  /// instance, JSON columns are automatically parsed and turned into their
  /// Dart representation. This means that a SQL column storing the JSON value
  /// `null` would get mapped to `null` in Dart, whereas it technically has a
  /// non-null value in SQL (`column IS NULL` would be false).
```

**Review by simolus3 — APPROVED** [review]
This is what I had in mind, looks great :+1:

**Comment by davidmartos96** [comment]
This would definitely solve the case for reading values from the database.

Regarding the original issue with JSON nulls, should a special value for that be added for the encoding phase?

**Comment by davidmartos96** [comment]
I'm specifically thinking about parameter binding, so that depending on a SQL literal in the query is not necessary

**Comment by simolus3** [comment]
> I'm specifically thinking about parameter binding, so that depending on a SQL literal in the query is not necessary

I think for symmetry I'd prefer a similar approach not needing a new type. I've explored that in #325, which adds an `isSqlNull` parameter to `TypedValue`.

**Pull request merged**